### ISSUE TITLE
I get this error on Mac when trying to install stubs

### ISSUE THREAD

**Issue by Strutsagget** [HvyIndustries/crane#122]
No handler found for the command: 'crane.downloadPHPLibraries'. An extension might be missing an activation event.

**Comment by Strutsagget** [comment]
It did work today when trying and wrote crane php stub.

**Comment by nevadascout** [comment]
Odd, I'll do some investigation

**Comment by Strutsagget** [comment]
I did update VSC on June releases. (the second one released today?)  Did not work(same error) on my colleague when i installed crane yesterday on his machine. on a fresh install of VSC.

**Comment by nevadascout** [comment]
Which version of OS X is the machine that's not working running?

**Comment by AbuJunayd** [comment]
I have the same error message on a fresh Win10 install, with fresh Visual Studio Code install.
Stubs won't install.

**Comment by AbuJunayd** [comment]
Forgot to mention that PHP Intellisense is also the 1st extension I installed and I actually didn't install any other.

**Comment by 0xNiklas** [comment]
Got the same error on a Windows 10 as @AbuJunayd, someone just got a fix or a workaround?

**Comment by nevadascout** [comment]
As a workaround you can download a zip of the stubs from [here](https://github.com/HvyIndustries/crane-php-stubs) and place them in the `C:/Users/[name]/Roaming/Crane/phpstubs/crane-php-stubs-master/` folder (you may have to create this folder)

Then press <kbd>F1</kbd>, type `reindex` and select the crane option to reindex the workspace.

**Comment by 0xNiklas** [comment]
Thanks dude! You're welcome

**Comment by 0xNiklas** [comment]
Getting this error on reindexing: `No handler found for the command: 'crane.rebuildSources'. An extension might be missing an activation event.`

**Comment by nevadascout** [comment]
You need to open a php file before trying to reindex the workspace

**Comment by vdoria** [comment]
Same issue

The workaround of @nevadascout is wrong place the files in %APPDATA%\Crane\phpstubs\crane-php-stubs-master\

%APPDATA% resolves to C:\Users\<username>\AppData\Roaming

The path likely won't exist so just use [START] +[R] > explorer %APPDATA% or easier just run the following command:

[START]+[R] > cmd > mkdir %APPDATA%\Crane\phpstubs\

Then download the file https://codeload.github.com/HvyIndustries/crane-php-stubs/zip/master (the zip contains the last folder as described above

After that you need to open a folder (your PHP project) and then [CTRL]+[SHIFT]+[R] to reindex crane. 

Finally PHP autocomplete for vscode :+1:

**Comment by nevadascout** [comment]
Oops, thanks for the correction @vdoria! I've updated my comment

**Comment by TheColorRed** [comment]
Also, the error message `No handler found for the command: 'crane.downloadPHPLibraries'` occurs because you haven't opened a php file to initialize the crane library. If you open a php file the crane library loads and the command then becomes executable.

**Comment by ichiriac** [comment]
We could fix this with activation events :

https://code.visualstudio.com/docs/extensionAPI/activation-events#_activationeventsoncommand

We could enable plugin for all exposed commands so there will no need to open a PHP file in order to make it work.

**Comment by TheColorRed** [comment]
I think you would have to use.

```json
"activationEvents": [
    "*"
]
```

However, would this attempt to start indexing files event if it isn't a php project?

**Comment by ichiriac** [comment]
@TheColorRed you are right, using '*' will trigger the plugin at start (vscode says it's not a good practice), but using `onCommand:crane.downloadPHPLibraries` will trigger it only if the command is requested by the user...

Instead using `*`, we could define all shortcut commands that depends on php environements.
